pkg/trtl/interceptor_test.go

```diff
@@ -0,0 +1,89 @@
+package trtl_test
+
+import (
+	"context"
+	"crypto/x509/pkix"
+	"net"
+
+	"github.com/trisacrypto/directory/pkg/trtl"
+	"github.com/trisacrypto/directory/pkg/trtl/pb/v1"
+	"github.com/trisacrypto/directory/pkg/trtl/peers/v1"
+)
+
+// TestPeerFromTLS tests that the peerFromTLS function is able to extract peer info
+// from TLS within the gRPC context.
+func (s *trtlTestSuite) TestPeerFromTLS() {
+	require := s.Require()
+
+	// TODO: We probably want to test more variations of the certificate.
+	expected := &trtl.PeerInfo{
+		Name: &pkix.Name{
+			Country:       []string{"US"},
+			Organization:  []string{"TRISA Development Client"},
+			Locality:      []string{"Raleigh"},
+			Province:      []string{"North Carolina"},
+			StreetAddress: []string{""},
+			PostalCode:    []string{""},
+			SerialNumber:  "", // TODO: Serial number was not parsed?
```

Comment:
Odd - may be something in the way we generated the certificate.

Reply:
Yeah, I expected it to be in there since it shows up on https://www.sslshopper.com/certificate-decoder.html

Reply:
Well then that's really strange …